beacon-chain/state/state-native/setters_balance_deposits.go

```diff
@@ -0,0 +1,63 @@
+package state_native
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state/state-native/types"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// AppendPendingBalanceDeposit is a mutating call to the beacon state to create and append a pending
+// balance deposit object on to the state. This method requires access to the Lock on the state and
+// only applies in electra or later.
+func (b *BeaconState) AppendPendingBalanceDeposit(index primitives.ValidatorIndex, amount uint64) error {
+	if b.version < version.Electra {
+		return errNotSupported("AppendPendingBalanceDeposit", b.version)
+	}
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	b.sharedFieldReferences[types.PendingBalanceDeposits].MinusRef()
+	b.sharedFieldReferences[types.PendingBalanceDeposits] = stateutil.NewRef(1)
```

Comment:
This looks odd. Why minus then just set it to one?

Reply:
That's how the shared state fields work with the garbage collector.

When you want to mutate a field that may be shared across copied states:
- Subtract your reference to the shared field
- Initialize a new field reference with count 1. This will be incremented when the state is copied in the future.
- When a field ref count reaches zero, then the garbage collector / finalizer allows the field to be destroyed

Reply:
Thanks!